client-spark/common/src/main/java/org/apache/spark/shuffle/handle/MutableShuffleHandleInfo.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.handle;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.client.PartitionDataReplicaRequirementTracking;
+import org.apache.uniffle.common.RemoteStorageInfo;
+import org.apache.uniffle.common.ShuffleServerInfo;
+import org.apache.uniffle.common.exception.RssException;
+import org.apache.uniffle.proto.RssProtos;
+
+/** This class holds the dynamic partition assignment for partition reassign mechanism. */
+public class MutableShuffleHandleInfo extends ShuffleHandleInfoBase {
```

Comment:
I'd suggest `ReplicaShuffleHandleInfo`, as this primarily provides the notion of replica, which are (secondary) mutable.

Reply:
This may not reflect the usage. How about FaultToleranceShuffleHandleInfo.

Reply:
A class should not be named after its usage, but after what it implements, as it has no knowledge about how it is being used, and it does not care, it provides only what it implements.

This handle info is **used** for fault tolerance, but there is no fault tolerance **built into** this class. It uses per-replica server infos to implement `getAvailablePartitionServersForWriter()` and `getAllPartitionServersForReader()`.

Reply:
> A class should not be named after its usage, but after what it implements, as it has no knowledge about how it is being used, and it does not care, it provides only what it implements.

Make sense. 

I have a concren that the `ReplicaShuffleHandleInfo` will make someone confused that they may think the  `SimpleShuffleHandleInfo` don't support replica mechanism.

What do you think? @jerqi

Reply:
Yes, I have the similar concern.